src/resolvers/Service.js

```diff
@@ -0,0 +1,207 @@
+       * retrieve with filter
+       */
+      let services = await Service.find(filter);
+      services.map((service) => estimatedServiceTimeConverter(service));
+
+      return services;
+    },
+    service: async (root, { id }, context, info) => {
+      /**
+       * - validate, disallow if provided id
+       * - allow all access
+       */
+
+      /**
+       * must have ID
+       */
+      if (id === null) {
+        throw UserInputError("No ID provided to search");
+      }
+
+      /**
+       * validate for existing
+       */
+      let existingService = await Service.findById(id);
+      if (existingService === null) {
+        throw UserInputError("no service with this ID");
+      }
+      existingService = estimatedServiceTimeConverter(existingService);
+
+      return existingService;
+    },
+  },
+  Mutation: {
+    createService: async (root, { serviceInput }, context, info) => {
+      /**
+       * - validate, disallow if provided id
+       * - allow all access
+       * - validate for existing name
+       */
+
+      /**
+       * disallow ID
+       */
+      if (serviceInput.id) {
+        throw UserInputError("Received id at create mutatiion");
+      }
+
+      /**
+       * validate for existing name
+       */
+      const existingService = await Service.findOne({
+        serviceNm: { $eq: serviceInput.serviceNm },
+      });
+      // suggest to format the name string like brandID + serviceName to enable same name for different brands
+      if (existingService) {
+        throw UserInputError("Name already taken");
+      }
+
+      /**
+       * create if passes all validations
+       */
+      let newService = await Service.create(serviceInput);
+      newService = estimatedServiceTimeConverter(newService);
+
+      return newService;
+    },
+    updateService: async (root, { serviceInput }, context, info) => {
+      /**
+       * - validate, disallow if provided id
+       * - allow manager and brand owner
+       */
+
+      /**
+       * destructure context user
+       */
+      const { type, employeeType } = context.user;
+
+      /**
+       * validate role
+       */
+      if (!roleValidator(type, employeeType)) {
+        throw new ApolloError(
+          "Only brand owner and manager can perform this operation"
+        );
+      }
+
+      /**
+       * must have ID
+       */
+      if (serviceInput.id === null) {
+        throw UserInputError("No ID provided to update");
+      }
+
+      /**
+       * validate for existing
+       */
+      let existingService = await Service.findById(serviceInput.id);
+      if (existingService === null) {
+        throw UserInputError("no service with this ID");
+      }
+      let service = await Service.findByIdAndUpdate(
+        serviceInput.id,
+        serviceInput,
+        { new: true }
+      );
+      service = estimatedServiceTimeConverter(service);
+
+      return service;
+    },
+    deleteService: async (root, { id }, context, info) => {
```

Comment:
delete option might not be available
if really need the should be Admin type user
Service is not service of a company/branch
as ref to the schema.jsonc
Service is the service our app will provide

Reply:
the role checking is at line 180

Reply:
I should update the comment at line 162 to brand owner only